src/Function/Packages.agda

```diff
@@ -0,0 +1,125 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Packages for types of functions
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Function.Packages where
+
+open import Function.Definitions
+open import Function.Structures
+open import Level using (_⊔_; suc)
+open import Relation.Binary using (Rel; Setoid)
+open import Relation.Binary.PropositionalEquality using (setoid)
+open Setoid using (Carrier; _≈_)
+
+------------------------------------------------------------------------
+-- Setoid packages
+
+record Injection {a b ℓ₁ ℓ₂} (From : Setoid a ℓ₁) (To : Setoid b ℓ₂)
+                 : Set (a ⊔ b ⊔ suc (ℓ₁ ⊔ ℓ₂)) where
+  field
+    f           : Carrier From → Carrier To
+    isInjection : IsInjection (_≈_ From) (_≈_ To) f
```

Comment:
The problem with taking a `Setoid` parameter is that the proofs that the relation is in an equality relation are all duplicated in both the `Setoid` and the `IsEqualityPreserving` proof inside `IsInjection`. I can't think of a neat way around this...

Reply:
Well we'd also have to add all the congruence proofs as well as the new injective, but I think that you're right and that's the correct way to go about doing things.

Reply:
Updated so they don't use the `IsX` structures which solves the problem of equality duplication.